### PR TITLE
fix(FEC-13348): dynamically adjust hotspot text with player size

### PR DESCRIPTION
**Pull request by lianbenjamin**
the requirement is to have the hotspot text fit into 1 line.

**solution:**
create a temporary hotspot container div and a temporary hotspot text div with the same styles as the real ones, make calculations to find the ideal font-size to use for the text.

Solves [FEC-13348](https://kaltura.atlassian.net/browse/FEC-13348)

[FEC-13348]: https://kaltura.atlassian.net/browse/FEC-13348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,4 +127,4 @@ If nvm installed: `nvm use` change version of current terminal to required.<br/>
 
 ### ARM Architecture support
 
-Install dependencies with `npm install --target_arch=x64` set target arch for running it through Rosetta (requires Rosetta installation).<br/>
\ No newline at end of file
+Install dependencies with `npm install --target_arch=x64` set target arch for running it through Rosetta (requires Rosetta installation).<br/>
```
Comment (semarche-kaltura) [inline README.md:130]: I think we don't need it anymore since repo uses `sass` instead of `node-sass`

File: src/components/Hotspot.tsx [hunk 2]
```diff
@@ -148,6 +151,43 @@ export default class Hotspot extends Component<Props, State> {
...
+    textEl.style.fontFamily = `${hotspot.styles['font-family']}`;
+    textEl.style.color = `${hotspot.styles['color']}`;
+    textEl.textContent = label || '';
+
+    document.body.appendChild(textEl);
+
+    const textWidth = textEl.clientWidth;
+    let initialFontSize = parseInt(hotspot.styles['font-size']);
+    let fontSizeToUse = initialFontSize;
+    const MINIMAL_FONT_SIZE = 10;
```
Comment (semarche-kaltura) [inline src/components/Hotspot.tsx:176]: let's move it to the top of file

File: src/components/Hotspot.tsx [hunk 3]
```diff
@@ -148,6 +151,43 @@ export default class Hotspot extends Component<Props, State> {
     }
   };
 
+  getFontSize = (layout: any, hotspot: any, label: string): number => {
+    let textEl = document.createElement('div');
```
Comment (semarche-kaltura) [inline src/components/Hotspot.tsx:155]: let's use Preact approach:
we can render div (without text, f.e. use 0 as default fontSize), get refs from it. Then once component did mount - make your calculation and set fontSize value to state.

**Review by semarche-kaltura — CHANGES_REQUESTED** [review]

**Review by MosheMaorKaltura — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (lianbenjamin) [reply README.md:130]: actually i have no idea why it is marked as a change, I didn't touch it and it doesn't seem like they are differ from each other

**Review by semarche-kaltura — APPROVED** [review]

**Pull request merged**